OrbitCore/Utils.h

```diff
@@ -275,6 +275,25 @@ class CWindowsMessageToString {
 };
 #endif
 
+//-----------------------------------------------------------------------------
+inline std::string ShortenStringWithEllipsis(const std::string& text, int max_len) {
+  if (max_len < 0) {
+    return text;
+  }
+  if (max_len <= 3) {
+    return text.length() <= 3 ? text : "...";
+  }
+  if (text.length() <= max_len) {
+    return text;
+  }
+
+  const size_t chars_to_cut =
+      text.length() - static_cast<uint32_t>(max_len) + 3;
+  const size_t l = (text.length() - chars_to_cut) / 2;
+  const size_t r = l + chars_to_cut;
+  return text.substr(0, l) + "..." + text.substr(r);
```

Comment:
If it's the intended behaviour to keep the rest of the string after the ellipsis, the name of the function will feel quite misleading to me. If I saw a function called `ShortenStringWithEllipsis` with a max_len parameter I would assume I get a string with `max_len` characters. But maybe I'm wrong and this is common practice in text processing?

Reply:
Ahhh. I'm sorry. I completely misread what it's doing. Please ignore this message. :see_no_evil: